sdk/nodejs/resource.ts

```diff
@@ -193,6 +273,33 @@ export abstract class Resource {
 
 (<any>Resource).doNotCapture = true;
 
+/**
+ * Alias is a partial description of prior named used for a resource. It can be processed in the context of a resource
+ * creation to determine what the full aliased URN would be.
+ */
+export interface Alias {
+    /**
+     * The previous name of the resource.
+     */
+    name?: Input<string>;
+    /**
+     * The previous type of the resource.
+     */
+    type?: Input<string>;
+    /**
+     * The previous parent of the resource.
+     */
+    parent?: Resource | Input<URN>;
```

Comment:
for the top three props, we should explain what it means when you don't pass anything in.

Reply:
Done.